src/main/java/frc/team2412/robot/Subsystems.java

```diff
@@ -5,23 +5,29 @@
 import frc.team2412.robot.subsystems.ArmSubsystem;
 import frc.team2412.robot.subsystems.DrivebaseSubsystem;
 import frc.team2412.robot.subsystems.IntakeSubsystem;
+import frc.team2412.robot.subsystems.VisionSubsystem;
 
 public class Subsystems {
 	public static class SubsystemConstants {
 		public static final boolean DRIVEBASE_ENABLED = true;
 		public static final boolean ARM_ENABLED = false;
 		public static final boolean INTAKE_ENABLED = false;
+		public static final boolean VISION_ENABLED = true;
 	}
 
 	public DrivebaseSubsystem drivebaseSubsystem;
 	public ArmSubsystem armSubsystem;
 	public IntakeSubsystem intakeSubsystem;
+	public VisionSubsystem visionSubsystem;
 
 	public Subsystems() {
 		boolean comp = Robot.getInstance().isCompetition();
 
 		if (DRIVEBASE_ENABLED) {
 			drivebaseSubsystem = new DrivebaseSubsystem();
+			if (VISION_ENABLED) {
+				visionSubsystem = new VisionSubsystem(drivebaseSubsystem::addVisionMeasurement);
+			}
```

Comment:
Can we do vision without drivebase

Reply:
I suppose when in sim, otherwise we'd be \*highly unlikely\* to not have a drivebase anyways. Probably still worth implementing, though there's a thorny issue since drivebase and us both depend on the pose estimator (so either it lives in one of us and the other depends on it, or it lives outside of both of us). Ideal solution would be the latter, but that could get large (also we'd probably change us conceptually from a subsystem to a sensor which is in the same place).

Reply:
We have decided to do this later if we have time, as right now we are focusing on accuracy and think that the effort it would take is not worth the benefit in this stage